Potion-client dependency outdated

The iloop-to-model service is failing with the latest version of iloop due to an issue fixed in potion-client v2.5.1.

```
[iloop-to-model-1]2017-05-05T09:09:17.725263194Z [2017-05-05 09:09:17 +0000] [15] [ERROR] Error handling request
[iloop-to-model-1]2017-05-05T09:09:17.725318530Z Traceback (most recent call last):
[iloop-to-model-1]2017-05-05T09:09:17.725328484Z   File "/usr/local/lib/python3.5/site-packages/aiohttp/web_server.py", line 62, in handle_request
[iloop-to-model-1]2017-05-05T09:09:17.725336720Z     resp = yield from self._handler(request)
[iloop-to-model-1]2017-05-05T09:09:17.725349157Z   File "/usr/local/lib/python3.5/site-packages/aiohttp/web.py", line 270, in _handle
[iloop-to-model-1]2017-05-05T09:09:17.725356950Z     resp = yield from handler(request)
[iloop-to-model-1]2017-05-05T09:09:17.725364297Z   File "/iloop-to-model/iloop_to_model/app.py", line 21, in wrapper
[iloop-to-model-1]2017-05-05T09:09:17.725372010Z     iloop = iloop_client(api, token)
[iloop-to-model-1]2017-05-05T09:09:17.725379224Z   File "/iloop-to-model/iloop_to_model/__init__.py", line 19, in iloop_client
[iloop-to-model-1]2017-05-05T09:09:17.725387020Z     verify=False
[iloop-to-model-1]2017-05-05T09:09:17.725394054Z   File "/usr/local/lib/python3.5/site-packages/potion_client/__init__.py", line 32, in __init__
[iloop-to-model-1]2017-05-05T09:09:17.725401717Z     self._fetch_schema()
[iloop-to-model-1]2017-05-05T09:09:17.725408894Z   File "/usr/local/lib/python3.5/site-packages/potion_client/__init__.py", line 43, in _fetch_schema
[iloop-to-model-1]2017-05-05T09:09:17.725422104Z     resource = self.resource_factory(name, resource_schema)
[iloop-to-model-1]2017-05-05T09:09:17.725429420Z   File "/usr/local/lib/python3.5/site-packages/potion_client/__init__.py", line 114, in resource_factory
[iloop-to-model-1]2017-05-05T09:09:17.725452870Z     for property_name, property_schema in schema['properties'].items():
[iloop-to-model-1]2017-05-05T09:09:17.725460014Z   File "/usr/local/lib/python3.5/site-packages/potion_client/resource.py", line 56, in __getitem__
[iloop-to-model-1]2017-05-05T09:09:17.725466887Z     return self._properties[item]
[iloop-to-model-1]2017-05-05T09:09:17.725473234Z KeyError: 'properties'
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,15 +1,38 @@
 sudo: required
 language: python
+
+branches:
+  only:
+    - master
+    - devel
+
+git:
+  depth: 3
+
 services:
-- docker
+  - docker
+
+env:
+  - DOCKER_COMPOSE_VERSION=1.12.0
+
 before_install:
-- sudo apt-get install procps
-- docker login -u=$DOCKER_USERNAME -p=$DOCKER_PASSWORD
-install: docker-compose up -d
+  - docker login -e=$DOCKER_EMAIL -u=$DOCKER_USERNAME -p=$DOCKER_PASSWORD
+  - sudo rm /usr/local/bin/docker-compose
+  - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+  - chmod +x docker-compose
+  - sudo mv docker-compose /usr/local/bin
+  - sudo apt-get install procps
+
+install:
+  - make start
+  
 script:
-- docker-compose exec web /bin/bash -c "pytest -v --cov=./iloop_to_model tests/"
+  - make test
+  - docker-compose logs
+  
 after_success:
-- docker-compose exec web /bin/bash -c "codecov --token $CODECOV_TOKEN"
+  - docker-compose exec web /bin/bash -c "codecov --token $CODECOV_TOKEN"
+
 deploy:
   provider: script
   script: ./push_to_hub_travis.sh

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,7 +1,13 @@
-FROM python:3.6
+FROM python:3.6-slim
 
-ADD requirements.txt requirements.txt
-RUN pip install -r requirements.txt
+RUN apt-get update && apt-get -y upgrade && \
+    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+
+RUN pip install --upgrade pip setuptools wheel
+COPY requirements.txt /tmp/requirements.txt
+RUN pip install --upgrade --process-dependency-links -r /tmp/requirements.txt && \
+    rm -rf /root/.cache /tmp/* /var/tmp/*
 
 ADD . ./iloop-to-model
 WORKDIR iloop-to-model

--- a/Makefile
+++ b/Makefile
@@ -1,0 +1,106 @@
+.PHONY: start stop clean test
+
+#################################################################################
+# GLOBALS                                                                       #
+#################################################################################
+
+PROJECT_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+
+
+#################################################################################
+# COMMANDS                                                                      #
+#################################################################################
+
+## Install and start the model service.
+start:
+	docker network inspect iloop-net || docker network create iloop-net
+	docker-compose up -d --build
+
+
+## Run the tests
+test: start
+	@echo "**********************************************************************"
+	@echo "* Running tests."
+	@echo "**********************************************************************"
+	docker-compose exec web /bin/bash -c "py.test -vxs --cov=./iloop_to_model tests/"
+
+
+## Shut down the Docker containers.
+stop:
+	docker-compose stop
+
+## Remove all containers.
+clean:
+	docker-compose down
+
+
+## Read the logs.
+logs:
+	docker-compose logs --tail="all" -f
+
+#################################################################################
+# PROJECT RULES                                                                 #
+#################################################################################
+
+
+
+#################################################################################
+# Self Documenting Commands                                                     #
+#################################################################################
+
+.DEFAULT_GOAL := show-help
+
+# Inspired by <http://marmelab.com/blog/2016/02/29/auto-documented-makefile.html>
+# sed script explained:
+# /^##/:
+# 	* save line in hold space
+# 	* purge line
+# 	* Loop:
+# 		* append newline + line to hold space
+# 		* go to next line
+# 		* if line starts with doc comment, strip comment character off and loop
+# 	* remove target prerequisites
+# 	* append hold space (+ newline) to line
+# 	* replace newline plus comments by `---`
+# 	* print line
+# Separate expressions are necessary because labels cannot be delimited by
+# semicolon; see <http://stackoverflow.com/a/11799865/1968>
+.PHONY: show-help
+show-help:
+	@echo "$$(tput bold)Available rules:$$(tput sgr0)"
+	@echo
+	@sed -n -e "/^## / { \
+		h; \
+		s/.*//; \
+		:doc" \
+		-e "H; \
+		n; \
+		s/^## //; \
+		t doc" \
+		-e "s/:.*//; \
+		G; \
+		s/\\n## /---/; \
+		s/\\n/ /g; \
+		p; \
+	}" ${MAKEFILE_LIST} \
+	| LC_ALL='C' sort --ignore-case \
+	| awk -F '---' \
+		-v ncol=$$(tput cols) \
+		-v indent=19 \
+		-v col_on="$$(tput setaf 6)" \
+		-v col_off="$$(tput sgr0)" \
+	'{ \
+		printf "%s%*s%s ", col_on, -indent, $$1, col_off; \
+		n = split($$2, words, " "); \
+		line_length = ncol - indent; \
+		for (i = 1; i <= n; i++) { \
+			line_length -= length(words[i]) + 1; \
+			if (line_length <= 0) { \
+				line_length = ncol - indent - length(words[i]) - 1; \
+				printf "\n%*s ", -indent, " "; \
+			} \
+			printf "%s ", words[i]; \
+		} \
+		printf "\n"; \
+	}' \
+	| more $(shell test $(shell uname) = Darwin && echo '--no-init --raw-control-chars')

--- a/README.md
+++ b/README.md
@@ -1,2 +1,32 @@
-[![Build Status](https://travis-ci.org/DD-DeCaF/iloop-to-model.svg?branch=master)](https://travis-ci.org/DD-DeCaF/iloop-to-model)
-[![Codecov](https://codecov.io/gh/DD-DeCaF/iloop-to-model/branch/master/graph/badge.svg)](https://codecov.io/gh/DD-DeCaF/iloop-to-model)
+# Iloop-to-model Service for DD-DeCaF
+
+|Build Status| |Coverage Status|
+
+
+.. |Build Status| image:: https://travis-ci.org/DD-DeCaF/iloop-to-model.svg?branch=master
+   :target: https://travis-ci.org/DD-DeCaF/iloop-to-model
+.. |Coverage Status| image:: https://codecov.io/gh/DD-DeCaF/iloop-to-model/branch/master/graph/badge.svg
+   :target: https://codecov.io/gh/DD-DeCaF/iloop-to-model
+
+## Installation
+
+Setup is as easy as typing `make start` in the project
+directory. However, the DD-DeCaF upload service depends on a running
+instance of the iloop backend. By default it expects a service
+`iloop-backend` to accept connections on port `80`, and
+`model-backend` on port 8000, on a docker network called
+`iloop-net`. You also need to set the `ILOOP_TOKEN` in your `.env`
+file (where you can also overide other variables) as this is specific
+for your specific user.
+
+
+| Variable                | Default Value                   | Description                                                                                                            |
+|-------------------------|---------------------------------|------------------------------------------------------------------------------------------------------------------------|
+| ``ILOOP_TO_MODEL_PORT`` | ``7000``                        | Exposed port of the upload service.                                                                                    |
+| ``ILOOP_API``           | ``http://iloop-backend:80/api`` | URL and port for the iloop service.                                                                                    |
+| ``ILOOP_TOKEN``         | ``''``                          | Token for the service to  connect to the iloop backend. (Not necessary if connecting  via the metabolica-ui-frontend.) |
+| ``MODEL_API``           | ``http://model-backend:80/api`` | URL port of the model service.                                                                                         |
+
+## Usage
+
+Type ``make`` in order to see all commonly used commands.

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1,12 +1,23 @@
-web:
-  build: .
-  dockerfile: Dockerfile
-  ports:
-    - 7000:7000
-  environment:
-    - PYTHONUNBUFFERED=1
-    - PYTHONASYNCIODEBUG=1
-    - ILOOP_TOKEN
-    - ILOOP_API
-    - MODEL_API
-  command: ["-w", "1", "-b", "0.0.0.0:7000", "-t", "150", "-k", "aiohttp.worker.GunicornWebWorker", "iloop_to_model.app:app"]
+version: "2.1"
+services:
+  web:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - "${ILOOP_TO_MODEL_PORT:-7000}:7000"
+    environment:
+      - PYTHONUNBUFFERED=1
+      - PYTHONASYNCIODEBUG=1
+      - ILOOP_TOKEN
+      - ILOOP_API=${ILOOP_API:-http://iloop-backend:80/api}
+      - MODEL_API=${MODEL_API:-http://model-backend:8000}
+    command: ["-w", "1", "-b", "0.0.0.0:7000", "-t", "150", "-k", "aiohttp.worker.GunicornWebWorker", "iloop_to_model.app:app"]
+    networks:
+      iloop-net:
+        aliases:
+          - iloop-to-model-backend
+
+networks:
+  iloop-net:
+    external: true

--- a/requirements.txt
+++ b/requirements.txt
@@ -11,5 +11,5 @@ uvloop
 pytest
 pytest-asyncio==0.5.0
 pytest-cov==2.4.0
-git+https://github.com/biosustain/venom.git@feat/json#egg=venom
+https://github.com/biosustain/venom/archive/feat/json.zip
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -2,7 +2,7 @@ aiohttp
 aiozmq
 aiohttp_cors
 msgpack-python
-potion_client
+potion_client>=2.5.1
 requests
 redis
 codecov
@@ -12,4 +12,3 @@ pytest
 pytest-asyncio==0.5.0
 pytest-cov==2.4.0
 https://github.com/biosustain/venom/archive/feat/json.zip
-

